src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -587,22 +587,23 @@ namespace smt::noodler {
                 conjuncts.push_back(init_aut_ass.get_lengths(var));
             }
         }
-        
+
 
         return LenNode(LenFormulaType::AND, conjuncts);
     }
 
     std::pair<LenNode, LenNodePrecision> DecisionProcedure::get_lengths() {
         LenNodePrecision precision = LenNodePrecision::PRECISE; // start with precise and possibly change it later
 
-        if (solution.length_sensitive_vars.empty() && this->not_contains.get_predicates().size() == 0 && !this->m_params.m_ca_constr) {
-            // There is not notcontains predicate to be solved and there are no length vars (which also means no disequations nor conversions), it is not needed to create the lengths formula.
-            // for the option m_ca_constr we completely skip this heuristic as there may remain also diseqations
-            return {LenNode(LenFormulaType::TRUE), precision};
+        if (solution.length_sensitive_vars.empty() && this->not_contains.get_predicates().empty() && !this->m_params.m_ca_constr) {
+            // There is not notcontains predicate to be solved and there are no length vars (which also means no
+            // disequations nor conversions), it is not needed to create the lengths formula.
+            return {preprocessing_len_formula, precision};
         }
```

Comment:
@vhavlena Hi, when solving `lan-rep/lan_replace0.smt2` there are no disequations coming in (`solution.length_sensitive_vars.empty() && this->not_contains.get_predicates().empty()` is `true`), so when we return either here, or on line 625 without actually doing anything anything in the diseq procedure, we terminate with UNSAT. I have checked that the procedure is trully not called (`assert(false)` on line 627.

Reply:
Why did you change `TRUE` to `preprocessing_len_formula `? During preprocessing you should not get an unsatisfiable length formula. The problem is that for the case of empty length vars there are some heuristics that basically rely on the fact `get_lenghts == TRUE`.